### ISSUE TITLE
[CDCSDK] Postgres crash seen in CDC + Tablet Splitting case with CDCSDK

### ISSUE THREAD

**Issue by shamanthchandra-yb** [yugabyte/yugabyte-db#16206]
Jira Link: [DB-5635](https://yugabyte.atlassian.net/browse/DB-5635)
### Description

http://stress.dev.yugabyte.com/stress_test/d6dd7665-4a3a-4e69-9ddf-ff179422959d

**Logs:**
[172.151.23.59.zip](https://github.com/yugabyte/yugabyte-db/files/10810447/172.151.23.59.zip) (Node1)
[172.151.25.4.zip](https://github.com/yugabyte/yugabyte-db/files/10810451/172.151.25.4.zip) (Node2)
[172.151.31.143.zip](https://github.com/yugabyte/yugabyte-db/files/10810452/172.151.31.143.zip) (Node 3)
[dumps.tar.gz.zip](https://github.com/yugabyte/yugabyte-db/files/10810459/dumps.tar.gz.zip)
[Docker_logs 2.zip](https://github.com/yugabyte/yugabyte-db/files/10810468/Docker_logs.2.zip)
[Framework_logs.zip](https://github.com/yugabyte/yugabyte-db/files/10810469/Framework_logs.zip)

**Backtrace of core dump:**

```
(lldb) target create "/home/yugabyte/yb-software/yugabyte-2.17.2.0-b181-centos-x86_64/postgres/bin/postgres" --core "/home/yugabyte/cores/core_postgres.20406"
Core file '/home/yugabyte/cores/core_postgres.20406' (x86_64) was loaded.
(lldb) bt all
* thread #1, name = 'postgres', stop reason = signal SIGABRT
  * frame #0: 0x00007f9afce520a7 libc.so.6`__GI_raise(sig=6) at raise.c:54
    frame #1: 0x00007f9afce534aa libc.so.6`__GI_abort at abort.c:89
    frame #2: 0x000055fdd57ca0fa postgres`errfinish(dummy=<unavailable>) at elog.c:815:3
    frame #3: 0x000055fdd57d293d postgres`elog_finish(elevel=<unavailable>, fmt=<unavailable>) at elog.c:1756:2
    frame #4: 0x000055fdd5609c0d postgres`LWLockQueueSelf(lock=<unavailable>, mode=<unavailable>) at lwlock.c:0
    frame #5: 0x000055fdd56095ea postgres`LWLockAcquire(lock=0x00007f9aeede9280, mode=LW_EXCLUSIVE) at lwlock.c:1222:3
    frame #6: 0x000055fdd555ff6d postgres`reaper [inlined] CleanupInvalidationStateInternal(segP=0x00007f9af8001980, proc=<unavailable>) at sinvaladt.c:337:2
    frame #7: 0x000055fdd555ff5e postgres`reaper [inlined] CleanupInvalidationStateForProc(proc=<unavailable>) at sinvaladt.c:387:2
    frame #8: 0x000055fdd555ff50 postgres`reaper [inlined] CleanupKilledProcess(proc=<unavailable>) at postmaster.c:3386:3
    frame #9: 0x000055fdd555ff1c postgres`reaper(postgres_signal_arg=<unavailable>) at postmaster.c:2945:5
    frame #10: 0x00007f9afd7d0ba0 libpthread.so.0`__restore_rt
    frame #11: 0x00007f9afcefe253 libc.so.6`__select at syscall-template.S:84
    frame #12: 0x000055fdd5562437 postgres`ServerLoop at postmaster.c:1732:13
    frame #13: 0x000055fdd555e4e5 postgres`PostmasterMain(argc=23, argv=0x000055fdd903c3c0) at postmaster.c:1431:11
    frame #14: 0x000055fdd546409f postgres`PostgresServerProcessMain(argc=23, argv=0x000055fdd903c3c0) at main.c:234:3
    frame #15: 0x000055fdd5133ca2 postgres`main + 34
    frame #16: 0x00007f9afce3f825 libc.so.6`__libc_start_main(main=(postgres`main), argc=23, argv=0x00007ffdc1b5f748, init=<unavailable>, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=0x00007ffdc1b5f738) at libc-start.c:289
    frame #17: 0x000055fdd5133bb9 postgres`_start at start.S:108

```

**Testcase:** test_cdc_with_tablet_splitting_cdcsdk

**Build:** 2.17.2.0-b181

[DB-5635]: https://yugabyte.atlassian.net/browse/DB-5635?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by qvad** [comment]
Have similar frames in https://github.com/yugabyte/yugabyte-db/issues/16278 w/o CDC

**Comment by timothy-e** [comment]
Duplicate of (and closed by) #18000
